src/sympc/module/nn/conv.py

```diff
@@ -0,0 +1,120 @@
+# stdlib
+from collections import OrderedDict
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Optional
+from typing import Tuple
+
+# third party
+import torch
+
+from sympc.tensor import MPCTensor
+
+
+class Conv2d:
+    __slots__ = [
+        "session",
+        "weight",
+        "bias",
+        "kernel_size",
+        "stride",
+        "padding",
+        "dilation",
+        "groups",
+    ]
+
+    in_channels: int
+    out_channels: int
+    kernel_size: Tuple[int, ...]
```

Comment:
Why are these defined but not set as `self.`, contrary to the next `stride`, `padding`... ?

Reply:
Hmm...I have seen that you could specify the type of the attributes, without them being bound to an instance like this (even if some of them are not used).
Should I remove them? (I do not have any strong opinion regarding this, I set those because I have seen them defined like this [in torch](https://github.com/pytorch/pytorch/blob/master/torch/nn/modules/conv.py#L56))

Reply:
I don't know to be honest! :)